3 reports.sql

```diff
@@ -0,0 +1,37 @@
+/*
+1)Inorder to see which flavor is most popular, can you show me how long has each flavor been 
+sitting on the shelf before it was sold (per chocolate flavor and chocolateType)? overall which flavor
+was on the shelf the longest? which the shortest?
+*/
+
+select c.FlavorDesc, c.ChocolateTypeDesc, AvgYearsOnShelf = 
+AVG(YEAR(c.SoldDate) - c.YearPutOnMarket), MinYearsOnShelfPerFlavor = MIN(YEAR(c.SoldDate) - c.YearPutOnMarket), MaxYearsOnShelfPerFlavor = MAX(YEAR(c.SoldDate) - c.YearPutOnMarket)
+from Chocolate c 
+group by c.FlavorDesc, c.ChocolateTypeDesc
+order by MinYearsOnShelfPerFlavor
+
+/*
+2)Can you print out a list as of January first 1917 which chocolate were put on the shelf but not yet sold.They got sold within the next 17 years.
+show the pieces in this order: Chocolate Name,yearPutOnShelf,DateSold,#Yearsold(the year it got sold after this point)
+*/
+
```

Comment:
this does not show the choc put on the shelf as of 1917.This is showing many more.

Reply:
Actually, as of January 1st, 1917 usually means everything before 1917 but not including - January is the first month of the year - unless you're a Navi - you wouldn't know what is going to be sold in that year on January 1st - only on December 31. Based on your question - I should not have had the = sign.
If you want to change your report to be the year of 1917 I could fix it.
I also slacked you about this - check it out ;)